lib/smack/SmackInstGenerator.cpp

```diff
@@ -623,6 +623,16 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst& ci) {
     Value* val = ci.getArgOperand(0);
     emit(Stmt::assign(rep->expr(&ci), rep->expr(val)));
 
+  } else if (SmackOptions::BitPrecise && SmackOptions::FloatEnabled &&
+    (name.find("llvm.convert.from.fp16.f32") != std::string::npos ||
+    name.find("llvm.convert.from.fp16.f64") != std::string::npos ||
+    name.find("llvm.convert.to.fp16.f32") != std::string::npos ||
+    name.find("llvm.convert.to.fp16.f64") != std::string::npos)) {
+    std::replace(name.begin(), name.end(), '.', '_');
```

Comment:
This way of generating function names is quite fragile. Maybe we could do something better here, like a constant map mapping LLVM intrinsic names into SMACK functions/models. Or maybe create a function `getIntrinsicModelName` that takes LLVM name and generated SMACK name for any intrinsic. Maybe that would be better.